src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -221,6 +225,11 @@ double RNN<OutputLayerType, InitializationRuleType>::Evaluate(
     targetSize = target.n_elem / rho;
     prevRho = rho;
   }
+  else if (inputSize == 0)
```

Comment:
If we initalize ``prevRho`` we have to check the ``targetSize`` here, since if someone calls ``Predict`` on an untrained model first and trains it afterwards ``inputSize`` is set but ``targetSize`` isn't.